tests/functional/commands/test_db_dump.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python
+"""Perform functional tests for db dump."""
+
+import os
+import tempfile
+
+import pytest
+
+import orion.core.cli
+from orion.core.io.database.pickleddb import PickledDB
+from orion.storage.base import setup_storage
+
+
+def execute(command, assert_code=0):
+    """Execute orion command and return returncode"""
+    returncode = orion.core.cli.main(command.split(" "))
+    assert returncode == assert_code
+
+
+def clean_dump(dump_path):
+    """Delete dumped files."""
+    for path in (dump_path, f"{dump_path}.lock"):
+        if os.path.isfile(path):
+            os.unlink(path)
+
+
+def test_default_storage(three_experiments_branch_same_name):
+    """Check default storage from three_experiments_branch_same_name"""
+    storage = setup_storage()
+    experiments = storage._db.read("experiments")
+    algos = storage._db.read("algo")
+    assert len(experiments) == 3
+    assert len(algos) == 3
+
+
+def test_dump_default(
+    three_experiments_branch_same_name_trials_benchmarks, capsys, testing_helpers
+):
+    """Test dump with default arguments"""
+    assert not os.path.exists("dump.pkl")
+    try:
+        execute("db dump")
+        assert os.path.isfile("dump.pkl")
+        dumped_db = PickledDB("dump.pkl")
+        testing_helpers.assert_tested_db_structure(dumped_db)
+    finally:
+        clean_dump("dump.pkl")
+
+
+def test_dump_overwrite(
+    three_experiments_branch_same_name_trials_benchmarks, capsys, testing_helpers
+):
+    """Test dump with overwrite argument"""
+    with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
You can avoid this line by using the fixture `tmp_path`: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
Done (here and for all comments below): https://github.com/Epistimio/orion/pull/1082/commits/91a10d9c2d70ebc27f73a2b8ba5a1ecfcf4634ea